go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -210,6 +210,42 @@ func (oc *BaseSecondaryNetworkController) Reconcile(netInfo util.NetInfo) error
 	return nil
 }
 
+func (oc *BaseSecondaryNetworkController) FilterOutResource(objType reflect.Type, obj interface{}) bool {
+	// Secondary networks that are not primary can potentially handle namespaces that are not a part of the networks
+	// NAD namespaces
+	if oc.IsPrimaryNetwork() {
+		return false
+	}
```

Comment:
Since this is bound to be used in more places, what about encapsulating this in a new utility method `CanServeNamespace(network NetInfo, ns string) bool` in `multinetwork.go`:

- If the network is non-default primary it only returns true if the provided namespace is among the NAD namespaces. Otherwise false.
- If the network is not primary just return true